buildenv/jenkins/common/build.groovy

```diff
@@ -830,7 +830,7 @@ def match_fail_pattern(outputLines) {
 
     println("Build failure, searching fail pattern in the last ${outputLines.size()} output lines")
 
-    for (failPattern in FAIL_PATTERN.tokenize(' ')) {
+    for (failPattern in FAIL_PATTERN.tokenize('|')) {
```

Comment:
While not strictly related to this change, `recompile()` below should throw the exception on the last retry even if the result is a similar failure. While you're here, could you fix that, please?

Reply:
Updated